jwk/helper.go

```diff
@@ -120,6 +127,25 @@ func FindKeyByPrefix(set *jose.JSONWebKeySet, prefix string) (key *jose.JSONWebK
 	return First(keys.Keys), nil
 }
 
+func FindPublicKey(set *jose.JSONWebKeySet) (key *jose.JSONWebKey, err error) {
+	keys := ExcludePrivateKeys(set)
+	if len(keys.Keys) == 0 {
+		return nil, errors.New("key not found")
+	}
+
+	return First(keys.Keys), nil
+}
+
+func ExcludePrivateKeys(set *jose.JSONWebKeySet) *jose.JSONWebKeySet {
+	keys := new(jose.JSONWebKeySet)
+	for _, k := range set.Keys {
+		if _, ok := k.Key.(*rsa.PrivateKey); !ok {
```

Comment:
This only works for RSA keys, but what if we use ECDSA? This should not exclude private keys, it should only include "allow listed" public keys.

Reply:
Let me check that @aeneasr